file_browser/src/components/ActionBar.tsx

```diff
@@ -0,0 +1,84 @@
+import { useMemo, useState } from "react";
+import { observer } from "mobx-react-lite";
+import fileSystemStore from "../stores/FileSystemStore";
+import { FileClass } from "../classes/File";
+import AddDialog from "./AddDialog";
+import RenameDialog from "./RenameDialog";
+import EditDialog from "./EditDialog";
+import s from "./ActionBar.module.scss";
+
+// icons
+import deleteIcon from "../assets/delete.png";
+import plus from "../assets/plus.png";
+import edit from "../assets/edit.png";
+import renameIcon from "../assets/rename.png";
+
+const ActionBar = observer(() => {
+
+  enum DialogState {
+    None = "none",
+    Rename = "rename",
+    Edit = 'edit',
+    Add = 'add'
+  }
```

Comment:
Textual values are only useful in 2 cases:
1. You need it to be human-redable as a displayed text.
2. You need it to be human-readable as JSON.
This is neither, so the textual values are redundant.